src/jit/importer.cpp

```diff
@@ -683,12 +677,18 @@ inline void Compiler::impAppendStmt(GenTree* stmt, unsigned chkLevel)
 //
 inline void Compiler::impAppendStmt(GenTree* stmt)
 {
-    // Point 'prev' at the previous node, so that we can walk backwards.
-    stmt->gtPrev = impTreeLast;
-
-    // Append the expression statement to the list.
-    impTreeLast->gtNext = stmt;
-    impTreeLast         = stmt;
+    if (impTreeList == nullptr)
```

Comment:
Have you checked if this change doesn't impact throughput?

Reply:
Yes, pin has not shown any impact (10 measures of crossgen System.Private.Corelib).